qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -15,222 +15,28 @@
-        and not passed to the fitter function.
+from .base_curve_analysis import BaseCurveAnalysis
+from .curve_data import CurveData, SeriesDef
 
-        .. code-block::
-
-            class AnalysisExample(CurveAnalysis):
 
-                __series__ = [
-                    SeriesDef(
-                        fit_func=lambda x, p0, p1, p2:
-                            exponential_decay(x, amp=p0, lamb=p1, baseline=p2),
-                    ),
-                ]
-
-                __fixed_parameters__ = ["p1"]
-
-        You can add arbitrary number of parameters to the class variable
-        ``__fixed_parameters__`` from the fit function arguments.
-        This parameter should be defined with the fit functions otherwise the analysis
-        instance cannot be created. In above example, parameter ``p1`` should be also
-        defined in the analysis options. This parameter will be excluded from the fit parameters
-        and thus will not appear in the analysis result.
-
-    Notes:
-        This CurveAnalysis class provides several private methods that subclasses can override.
-
-        - Customize pre-data processing:
-            Override :meth:`~self._format_data`. For example, here you can apply smoothing
-            to y values, remove outlier, or apply filter function to the data.
-            By default, data is sorted by x values and the measured values at the same
-            x value are averaged.
-
-        - Create extra data from fit result:
-            Override :meth:`~self._extra_database_entry`. You need to return a list of
-            :class:`~qiskit_experiments.framework.analysis_result_data.AnalysisResultData`
-            object. This returns an empty list by default.
-
-        - Customize fit quality evaluation:
-            Override :meth:`~self._evaluate_quality`. This value will be shown in the
-            database. You can determine the quality represented by the predefined string
-            "good" or "bad" based on fit result,
-            such as parameter uncertainty and reduced chi-squared value.
-            This returns ``None`` by default. This means evaluation is not performed.
-
-        - Customize fitting options:
-            Override :meth:`~self._generate_fit_guesses`. For example, here you can
-            calculate initial guess from experiment data and setup fitter options.
-
-        See docstring of each method for more details.
-
-        Note that other private methods are not expected to be overridden.
-        If you forcibly override these methods, the behavior of analysis logic is not well tested
-        and we cannot guarantee it works as expected (you may suffer from bugs).
-        Instead, you can open an issue in qiskit-experiment github to upgrade this class
-        with proper unittest framework.
-
-        https://github.com/Qiskit/qiskit-experiments/issues
-    """
+class CurveAnalysis(BaseCurveAnalysis):
+    """Base class for curve analyis."""
```

Comment:
This is still base class (e.g. `ErrorAmplificationAnalysis(CurveAnalysis)`). The reason we introduce `BaseCurveAnalysis` is to remove boilerplate code in `GroupedCurveAnalysis`.

Reply:
hmmm the naming gets a bit confusing no? Since `CurveAnalysis` is still a base class it could be called `BaseCurveAnalysis` which is already taken. What are the distinctive features between `CurveAnalysis` (a base class) and `BaseCurveAnalysis`? Perhaps that can help with the naming.

Reply:
Can you also add a bit more to the class docstring to explain why we have this class? Currently it reads `Base class for curve analyis.` which is too brief given that we have another class called `BaseCurveAnalysis`.

Reply:
This will be clearer once we introduce the `GroupCurveAnalysis` (perhaps `MultiGroupCurveAnalysis` to be precise). The base one is the superclass of both `CurveAnalysis` and `GroupCurveAnalysis `. The difference between `CurveAnalysis` and `BaseCurveAnalysis` is implementation of `_run_analysis` method. In the group one the logic inside the `_run_analysis` will be different from `CurveAnalysis` though each subroutines can be reused. This is why `BaseCurveAnalysis` only implements subroutines.